static/report/js/compare.js

```diff
@@ -0,0 +1,637 @@
+  var dimensions = {height: null, width: null };
+  var margin = {top: 20, right: 50, bottom: 20, left: 50};
+  var xAxis, yAxis;
+  var dom_width, dom_height;
+  var width, height;
+  var root = { };
+  var dots = { };
+  var chart = { };
+
+  function my ( ) {
+    // dots = chart.selectAll('circle')
+    // dots.enter( )
+      // .append("circle")
+    // console.log('data', my.data);
+    var selection = dots.selectAll('.dots').data(my.data || [ ]);
+    // console.log('dots', dots, selection);
+    selection.enter( ).append('circle').attr('class', 'dots');
+    selection.exit( ).remove( );
+    selection.call(render_circles)
+    // .exit( ).remove( );
+  }
+
+  function make_x_axis ( ) {
+    return d3.svg.axis( )
+      .scale(scales.x)
+      ;
+  }
+
+  function make_y_axis ( ) {
+    return d3.svg.axis()
+      .scale(scales.y)
+      ;
+  }
+
+  function gridlines (selection) {
+    selection.selectAll('.x.gridlines')
+      // .attr("transform", "translate(0," + height + ")")
+      .attr("transform", "translate(" + margin.left + "," + height + ")")
+      // .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      .call(make_x_axis( )
+        .tickSize(-height, 0, 0)
+        .tickFormat("")
+      )
+            
+    selection.selectAll('.y.gridlines')
+      .attr("transform", "translate(" + margin.left + ", " + (margin.top) + ")")
+      .call(make_y_axis( )
+        .tickSize(-width , 0, 0)
+        .tickValues([40, 80, 120,  180, 240, 300,  400])
+        .tickFormat("")
+        .orient("left")
+        // .ticks(6)
+      )
+
+  }
+  function render_circles (dots) {
+    dots
+        .transition( )
+        .attr("r", 5)
+        .attr("title",  function (d) { return d.key; })
+        .attr("fill", function (d) { return d.values.color; })
+        .attr("cx", function (d) { return scales.x(d.values.target); })
+        .attr("cy", function (d) { return scales.y(d.values.mean); })
+    return dots;
+  }
+
+  function render (selection) {
+    console.log('selection');
+  }
+
+  my.setRange = function _setRange (begin, end) {
+    adjuster(begin, end);
+  }
+
+  function adjust_range (begin, end) {
+    my.page.refresh(begin, end);
+  }
+
+  var adjuster = _.debounce(adjust_range, 1000);
+
+  function on_data (payload) {
+    var days = time_in_range(payload, opts);
+    my.data = days;
+    my( );
+    // console.log('got days', days);
+  }
+
+  function init ( ) {
+    my.page = pager({begin: opts.begin, end: opts.end, callback: on_data, cache: opts.cache});
+    get_dimensions( );
+
+    scales.x = d3.scale.linear( )
+      .domain([0, 100])
+      .range([0, width])
+      ;
+
+    scales.y = d3.scale.log( )
+          .domain( [40, 400] )
+          .rangeRound( [height - (margin.top + margin.bottom), 1] )
+          .base(12)
+      ;
+
+
+    // Axis
+    xAxis = make_x_axis( )
+      .ticks(10)
+      .tickFormat(function(d) { return parseInt(d, 10) + "%"; })
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2165148169/issues/source?bid=2452379&fileBranchId=2967311#l549)